packages/tpetra/core/src/Tpetra_BlockCrsMatrix_def.hpp

```diff
@@ -2945,7 +2945,12 @@ class GetLocalDiagCopy {
     // We use a "struct of arrays" approach to packing each row's
     // entries.  All the column indices (as global indices) go first,
     // then all their owning process ranks, and then the values.
-    exports.resize (totalNumBytes);
+    if(exports.extent(0) != totalNumBytes)
+    {
+      const std::string oldLabel = exports.d_view.label ();
+      const std::string newLabel = (oldLabel == "") ? "exports" : oldLabel;
+      exports = Kokkos::DualView<packet_type*, buffer_device_type>(newLabel, totalNumBytes);
```

Comment:
Thanks Brian!  This looks good.  In the future, we should think about allocating without initializing, but this is fine for now.